src/org/zaproxy/zap/extension/proxies/DialogAddProxy.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.proxies;
+
+import java.awt.Dialog;
+
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.view.AbstractFormDialog;
+
+class DialogAddProxy extends AbstractFormDialog {
+
+    private static final long serialVersionUID = 4460797449668634319L;
+
+    private static final String DIALOG_TITLE = Constant.messages.getString("options.proxy.dialog.proxy.add.title");
+
+    private static final String CONFIRM_BUTTON_LABEL = Constant.messages
+            .getString("options.proxy.dialog.proxy.add.button.confirm");
+
+    protected OptionsLocalProxyPanel proxyPanel;
+
+    protected ProxiesParamProxy proxy;
+
+    private ExtensionProxies extension;
+
+    public DialogAddProxy(Dialog owner, ExtensionProxies extension) {
+        super(owner, DIALOG_TITLE);
+        this.extension = extension;
+        setConfirmButtonEnabled(true);
+    }
+
+    protected DialogAddProxy(Dialog owner, String title, ExtensionProxies extension) {
+        super(owner, title);
+        this.extension = extension;
+    }
+
+    @Override
+    protected JPanel getFieldsPanel() {
+        if (proxyPanel == null) {
+            proxyPanel = new OptionsLocalProxyPanel();
+        }
+        return proxyPanel;
+    }
+
+    @Override
+    protected String getConfirmButtonLabel() {
+        return CONFIRM_BUTTON_LABEL;
+    }
+
+    @Override
+    protected void init() {
+        proxy = null;
+    }
+
+    @Override
+    protected boolean validateFields() {
+        ProxiesParamProxy testProxy = proxyPanel.getProxy();
+
+        if (extension.getAdditionalProxy(testProxy.getAddress(), testProxy.getPort()) != null) {
```

Comment:
This case is actually caught by line 90: extension.canListenOn(testProxy.getAddress(), testProxy.getPort()
that ok?

Reply:
Ideally this should be validated against the "latest" address/port, for example, if you change the main proxy and try to add an alternative proxy with the same port it will say that it already exists (even though in the options it has a different port).

Reply:
I think that `this.proxy` only contains the currently modified proxy. And in line 90 it is checked if that proxy was modified, isn't it? 
But I meant that the new proxy should be validated against the other proxies (MainProxy and AdditionalProxies) and I mean not the already listening ones, I mean the newly configured proxies.